File: Tests/Types/AnnotationsNullable/Output/self-referencing-type.ts
/* eslint-disable */
// tslint:disable

export class SelfReferencingType {
    public stringProperty?: string | undefined;
    public selfProperty: SelfReferencingType;
    public selfList?: SelfReferencingType[] | undefined;

    public constructor(init?: Partial<SelfReferencingType>) {
        Object.assign(this, init);
    }
}

// outputid:f13dd313-1bd6-4a8e-9b3f-d6266d3a25ff

File: Tests/v10/SelfReferencingAnnotationsNullableEnabled/SelfReferencingType.cs
using KY.Generator;

namespace SelfReferencing;

[GenerateTypeScriptModel]
public class SelfReferencingType
{
    public string StringProperty { get; set; } = "";
    public SelfReferencingType? SelfProperty { get; set; }
    public List<SelfReferencingType> SelfList { get; set; } = [];
}

